src/mixins/README.md

```diff
@@ -0,0 +1,82 @@
+# Mixins
+
+This directory contains mixins for `Application`.
+
+## Overview
+
+A mixin is a class which take the input of a base class and adds / modifies existing functions / status values and returns a new class which can be instantiated. The type of the returned class will still be the base class. Mixins let you extend the BaseClass by adding more functions to it, overriding existing ones or modifiying their behavior.
+
+## Basic Usage
+
+The general idea behind using a mixin is to pass in the BaseClass to the Mixin and create a new class. 
+
+**Example**
+```
+cosnt newClass = MixinClass(BaseClass);
```

Comment:
typo, should be `const`

I think we should be recommending `class`-based approach in our examples.

```ts
class MyApp extends MixinClass(Application) {
}
```

One of the downsides of your current example is that we have a class constructor stored in a variable with camelCase name, when the convention for class constructors is to use PascalCase names.

(Same comments apply to the example below.)